manifests/service/pam.pp

```diff
@@ -7,27 +7,49 @@
 #
 # == Trevor Vaughan <mailto:tvaughan@onyxpoint.com>
 #
-class sssd::service::pam (
-  Optional[String]             $description                    = undef,
-  Optional[Sssd::DebugLevel]   $debug_level                    = undef,
-  Boolean                      $debug_timestamps               = true,
-  Boolean                      $debug_microseconds             = false,
-  Integer                      $reconnection_retries           = 3,
-  Optional[String]             $command                        = undef,
-  Integer                      $offline_credentials_expiration = 0,
-  Integer                      $offline_failed_login_attempts  = 3,
-  Integer                      $offline_failed_login_delay     = 5,
-  Integer                      $pam_verbosity                  = 1,
-  Integer                      $pam_id_timeout                 = 5,
-  Integer                      $pam_pwd_expiration_warning     = 7,
-  Optional[Integer]            $get_domains_timeout            = undef,
-  Optional[String]             $pam_trusted_users              = undef,
-  Optional[String]             $pam_public_domains             = undef
-) {
+# == Parameters available to this template
+# command
+# debug_level
+# debug_timestamps
+# debug_microseconds
+# description
+# get_domains_timeout
+# offline_credentials_expiration
+# offline_failed_login_attempts
+# offline_failed_login_delay
+# pam_id_timeout
+# pam_pwd_expiration_warning
+# pam_public_domains
+# pam_trusted_users
+# pam_verbosity
+# reconnection_retries
+
+class sssd::service::pam {
+
   include '::sssd'
 
+  # These varaibles are referenced inside the autofs template, and
+  # because we don't want to worry about scope inside of the template
+  # we handle it here.
+
+  $description                    = $sssd::description
```

Comment:
Are we really going to bloat `init.pp` endlessly for each of these subsystems? What happens when two parameters overlap (like they currently do between this and `nss`?